common/src/main/java/com/wynntils/screens/chattabs/widgets/ChatTabsWidget.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.chattabs.widgets;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Services;
+import com.wynntils.core.text.StyledText;
+import com.wynntils.screens.chattabs.ChatTabEditingScreen;
+import com.wynntils.services.chat.ChatTab;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.network.chat.Component;
+
+public class ChatTabsWidget extends AbstractWidget {
+    private final float gridDivisions;
+    private final ChatTab chatTab;
+    private final Button deleteButton;
+    private final Button moveUpButton;
+    private final Button moveDownButton;
+    private final ChatTabEditingScreen parent;
+
+    public ChatTabsWidget(
+            float x,
+            float y,
+            int width,
+            int height,
+            ChatTab chatTab,
+            float gridDivisions,
+            ChatTabEditingScreen parent) {
+        super((int) x, (int) y, width, height, Component.literal(chatTab.getName()));
+
+        this.chatTab = chatTab;
+        this.gridDivisions = gridDivisions;
+        this.parent = parent;
+
+        this.deleteButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.chatTabsGui.delete")
+                                .withStyle(ChatFormatting.DARK_RED),
+                        (button) -> Services.ChatTab.removeTab(chatTab))
+                .pos((int) (this.getX() + (this.width / this.gridDivisions * 17)), this.getY() + (this.height / 2) - 10)
```

Comment:
this calculation is a bit difficult to follow. can you extract some parts of this to vars (like `this.width / this.gridDivisions`

Reply:
arguably that would result in weirder code